### PR TITLE
Update the 401 error solution

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: docs/tsg.md [hunk 1]
```diff
@@ -106,17 +106,28 @@ For ASP.NET Core SignalR's other transport type, SSE and longpolling, this means
...
             {
-                // extend the TTL
-                options.AccessTokenLifetime = TimeSpan.FromDays(1);
-            });
+                await Task.Delay(new Random().Next(0,5) * 1000);
+                await connection.StartAsync();
+            };
+```
+
+Sample code for ASP.NET JavaScript Client is:
```
Comment (chenkennt) [inline docs/tsg.md]: What about ASP.NET Core Javascript client?

File: docs/tsg.md [hunk 2]
```diff
@@ -106,17 +106,28 @@ For ASP.NET Core SignalR's other transport type, SSE and longpolling, this means
...
 
-There is a `AccessTokenLifetime` for you to customize the lifetime of the access token.
+For security concerns, extend TTL is not encouraged. We suggest adding reconnect logic from client to restart the connection when such 401 occurs. When client restarts the connection, it will negotiate with app server to get the JWT token again and get a renewed token.
+
+Sample code for ASP.NET Core supporting restarting the connection is:
 
 ```cs
-services.AddSignalR()
-        .AddAzureSignalR(options =>
+connection.Closed += async (error) =>
```
Comment (chenkennt) [inline docs/tsg.md]: This is not a complete sample for reconnect.

Comment (chenkennt) [reply docs/tsg.md]: Let's create a section for reconnect somewhere and reference it from here.

File: docs/tsg.md [hunk 3]
```diff
@@ -117,39 +117,60 @@ For ASP.NET Core SignalR's other transport type, SSE and longpolling, this means
...
+
+4. When SignalR Service has some internal errors such as instance restart, failover..
 
 ### Troubleshooting Guide
 1. Open app server side log to see if anything abnormal took place
 2. Check app server side event log to see if the app server restarted
 3. Create an issue to us providing time frame, and email the resource name to us
 
+<a id="server-conn-drop"/>
 ## ⌛️[TODO]Server side connection drop
```
Comment (JialinXin) [inline docs/tsg.md]: >side [](start = 18, length = 4)

remove side as well?

File: docs/tsg.md [hunk 4]
```diff
@@ -117,39 +117,60 @@ For ASP.NET Core SignalR's other transport type, SSE and longpolling, this means
...
-Possibility 1. App server restarts
-Possibility 2. **ASRS**(**A**zure **S**ignal**R** **S**ervice) internal error
+Client connections can drop under various circumstances:
+1. When `Hub` throws exceptions with the incoming request.
+
+2. When the server connection it routed to drops, see below section for details on [server connection drops](#server-conn-drop).
+
+3. When network connectivity issue happens between client and SignalR Service 
+
+4. When SignalR Service has some internal errors such as instance restart, failover..
```
Comment (JialinXin) [inline docs/tsg.md]: >. [](start = 84, length = 1)

extra .?

File: docs/tsg.md [hunk 5]
```diff
@@ -117,39 +117,60 @@ For ASP.NET Core SignalR's other transport type, SSE and longpolling, this means
...
-2. Client side error log: "Service timeout. 30.00ms elapsed without receiving a message from service."
+* [ASP.NET Core JavaScript Client](../samples/ChatSample/ChatSample/wwwroot/index.html#L164)
+
+* [ASP.NET Core C# Client](../samples/AspNet.ChatSample/AspNet.ChatSample.CSharpClient/Program.cs#L78)
+
+* [ASP.NET Core JavaScript Client](../samples/AspNet.ChatSample/AspNet.ChatSample.JavaScriptClient/wwwroot/index.html#L71)
+
+## ⌛️[TODO]Client connection drop
+
+When client is connected to the Azure SignalR, the persistent connection between client and Azure SignalR can sometimes drop for different reasons. This section describes several possibilities causing such connection drop, and provide some guidence to how to identify the root cause.
```
Comment (JialinXin) [inline docs/tsg.md]: >provide [](start = 227, length = 7)

provides

Comment (JialinXin) [inline docs/tsg.md]: >guidence [](start = 240, length = 8)

guidance

File: docs/tsg.md [hunk 6]
```diff
@@ -117,39 +117,60 @@ For ASP.NET Core SignalR's other transport type, SSE and longpolling, this means
...
+
+As the connections between app server and SignalR Service are persistent connections, they may experience network connectivity issues. In the Server SDK, we have “Always Reconnect” strategy to server connections. As the best practice, we also encourage users to add continuous reconnect logic to the clients with a random delay time to avoid massive simultaneous requests to the server.
+
+On regular basis there are new version releases for the Azure SignalR Service, and sometimes the Azure wide OS patching or upgrades or occasionally our dependent services’ interruption. These may bring in a very short period of service disruption, but as long as client side has the disconnect/reconnect mechanism, the impact is minimal like any client-side caused disconnect-reconnect.
+
+This section describes several possibilities leading to server connection drop, and provide some guidence to how to identify the root cause.
```
Comment (JialinXin) [inline docs/tsg.md]: >provide some guidence [](start = 84, length = 21)

provides, guidance

File: docs/tsg.md [hunk 7]
```diff
@@ -117,39 +117,60 @@ For ASP.NET Core SignalR's other transport type, SSE and longpolling, this means
...
+2. `Service timeout. 30.00ms elapsed without receiving a message from service.`
+3. `{"type":7,"error":"Connection closed with an error."}`
+4. `{"type":7,"error":"Internal server error."}`
 
 ### Root cause:
-Possibility 1. App server restarts
-Possibility 2. **ASRS**(**A**zure **S**ignal**R** **S**ervice) internal error
+Client connections can drop under various circumstances:
+1. When `Hub` throws exceptions with the incoming request.
+
```
Comment (JialinXin) [inline docs/tsg.md]: empty extra lines.

**Pull request merged**